src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java

```diff
@@ -58,104 +59,40 @@ public Event findById(Long id) {
         try {
             return jdbcTemplate.queryForObject(Queries.SQL_SELECT_EVENT_BY_ID, eventRowMapper, id);
         } catch (EmptyResultDataAccessException e) {
-            throw eventEntityNotFoundException(e.getMessage(), "id = " + id);
+            throw new EventNotFoundException("Failed to obtain event during `select` {id = " + id + "}, EventDao.findById", e);
         } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "id = " + id);
+            throw new CRUDException("Error during `select` {id = " + id + "}, EventDao.findById", e);
         }
     }
 
-    @Override
-    public Event findByTransactionId(Long transactionId) {
+    public List<Event> findAll(Map<String, ?> params) {
+        final String where = Stream
+            .of("account_id", "warehouse_id", "author_id", "type", "date", "after", "before")
+            .filter(params::containsKey)
+            .map(x -> buildSqlFilterCondition(x, params.get(x)))
+            .collect(Collectors.joining("\n and "));
+        final String query = String.format("select * from events where %s",
+            where.isBlank() ? "TRUE" : where);
```

Comment:
Need to think about pagination, please do not forget to add this when you will work with UI and controllers